pennylane/operation.py

```diff
@@ -1950,11 +1979,13 @@ def _queue_category(self):
         return "_ops" if isinstance(self, Operation) else None
 
     @property
-    def is_hermitian(self):
+    def is_hermitian(self) -> bool:
         """All observables must be hermitian"""
         return True
 
-    def __matmul__(self, other):
+    def __matmul__(
+        self, other: Union["Operator", "Tensor", "qml.ops.Hamiltonian", "qml.ops.LinearCombination"]
```

Comment:
Given they are all instances of type `Operator`, do we need to specify different types of special operator?